registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/test/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowActionJobTest.java

```diff
@@ -0,0 +1,197 @@
+		public MosipEventBus getEventBus(Object verticleName, String url, int instanceNumber) {
+			vertx = Vertx.vertx();
+
+			return new MosipEventBus() {
+
+				@Override
+				public Vertx getEventbus() {
+					return vertx;
+				}
+
+				@Override
+				public void consume(MessageBusAddress fromAddress,
+						EventHandler<EventDTO, Handler<AsyncResult<MessageDTO>>> eventHandler) {
+
+				}
+
+				@Override
+				public void consumeAndSend(MessageBusAddress fromAddress, MessageBusAddress toAddress,
+						EventHandler<EventDTO, Handler<AsyncResult<MessageDTO>>> eventHandler) {
+
+				}
+
+				@Override
+				public void send(MessageBusAddress toAddress, MessageDTO message) {
+
+				}
+			};
+		}
+
+		@Override
+		public void send(MosipEventBus mosipEventBus, MessageBusAddress toAddress, MessageDTO message) {
+		}
+	};
+
+	@Mock
+	RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	@Mock
+	private AuditLogRequestBuilder auditLogRequestBuilder;
+
+	@Mock
+	private LogDescription description;
+
+	@Mock
+	WorkflowActionService workflowActionService;
+
+	@Before
+	public void setup() throws Exception {
+
+		ReflectionTestUtils.setField(workflowActionJob, "fetchSize", 2);
+		ReflectionTestUtils.setField(workflowActionJob, "elapseTime", 21600);
+		Field auditLog = AuditLogRequestBuilder.class.getDeclaredField("registrationProcessorRestService");
+		auditLog.setAccessible(true);
+		@SuppressWarnings("unchecked")
+		RegistrationProcessorRestClientService<Object> mockObj = Mockito
+				.mock(RegistrationProcessorRestClientService.class);
+		auditLog.set(auditLogRequestBuilder, mockObj);
+		AuditResponseDto auditResponseDto = new AuditResponseDto();
+		ResponseWrapper<AuditResponseDto> responseWrapper = new ResponseWrapper<>();
+		responseWrapper.setResponse(auditResponseDto);
+		auditLogRequestBuilder.createAuditRequestBuilder("test case description", EventId.RPR_401.toString(),
+				EventName.ADD.toString(), EventType.BUSINESS.toString(), "1234testcase", ApiName.AUDIT);
+	}
+
+	@Test
+	public void testProcessSuccess() throws TablenotAccessibleException, WorkflowActionException {
+
+		List<InternalRegistrationStatusDto> actionablePausedPacketList = new ArrayList<>();
+		InternalRegistrationStatusDto registrationStatusDto = new InternalRegistrationStatusDto();
+
+		registrationStatusDto.setRegistrationId("2018701130000410092018110735");
+		registrationStatusDto.setRegistrationStageName("PacketValidatorStage");
+		registrationStatusDto.setDefaultResumeAction("RESUME_PROCESSING");
+		registrationStatusDto.setResumeTimeStamp(LocalDateTime.now());
+		registrationStatusDto.setReProcessRetryCount(0);
+		registrationStatusDto.setLatestTransactionStatusCode(RegistrationTransactionStatusCode.REPROCESS.toString());
+
+		actionablePausedPacketList.add(registrationStatusDto);
+		Mockito.when(registrationStatusService.getActionablePausedPackets(anyInt()))
+				.thenReturn(actionablePausedPacketList);

```

Comment:
Ideally this test case should give a combination of registrationStatusDto with all types of resume actions and see if the processWorkflowAction is called those many time with proper dtos list on each different resume actions

Reply:
fixed